### PR TITLE
feat: add tidb-ctl trigger

### PR DESCRIPTION
**Pull request by lijie0123**
# Why:
- tidb-ctl only build for master branch

### CODE REVIEW

File: apps/prod/tekton/configs/triggers/triggers/kustomization.yaml [hunk 1]
```diff
@@ -24,6 +24,8 @@ resources:
   - pingcap/tidb-dashboard/fake-git-push.yaml
   - pingcap/tidb-dashboard/git-create-tag.yaml
   - pingcap/tidb-dashboard/git-push.yaml
+  - pingcap/tidb-ctl/git-create-tag.yaml
+  - pingcap/tidb-ctl/git-push.yaml
```
Comment (wuhuizuo) [inline apps/prod/tekton/configs/triggers/triggers/kustomization.yaml:28]: please order the list.

File: apps/prod/tekton/configs/triggers/triggers/pingcap/tidb-ctl/git-push.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        - name: filter
+          value: >-
+            body.repository.owner.login == 'pingcap' && body.repository.name == 'tidb-ctl'
+    - name: filter on branches
+      ref:
+        name: cel
+      params:
+        - name: filter
+          value: >-
+            body.ref.matches('^refs/heads/(master)$')
```
Comment (wuhuizuo) [inline apps/prod/tekton/configs/triggers/triggers/pingcap/tidb-ctl/git-push.yaml:24]: Please combine them for performance.

File: apps/prod/tekton/configs/triggers/triggers/pingcap/tidb-ctl/git-create-tag.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        - name: filter
+          value: >-
+            body.repository.owner.login == 'pingcap' && body.repository.name == 'tidb-ctl'
+    - name: filter on version tags
+      ref:
+        name: cel
+      params:
+        - name: filter
+          value: >-
+            body.ref.matches('^v[0-9]+[.][0-9]+[.][0-9]+')
```
Comment (wuhuizuo) [inline apps/prod/tekton/configs/triggers/triggers/pingcap/tidb-ctl/git-create-tag.yaml:24]: please combine them for performance

File: apps/prod/tekton/configs/triggers/triggers/pingcap/tidb-ctl/git-push.yaml (hunk 2 above)
Comment (lijie0123) [reply apps/prod/tekton/configs/triggers/triggers/pingcap/tidb-ctl/git-push.yaml]: done

File: apps/prod/tekton/configs/triggers/triggers/kustomization.yaml (hunk 1 above)
Comment (lijie0123) [reply apps/prod/tekton/configs/triggers/triggers/kustomization.yaml]: done

**Review by wuhuizuo — APPROVED** [review]

**Pull request merged**